### PR TITLE
Php82 fixes

### PR DESCRIPTION
**Pull request by surrim**
I fixed the most critical things in order to get S9Y working on PHP8.2.

I had to update Smarty to [v4.3.0](https://github.com/smarty-php/smarty/releases/tag/v4.3.0) in order to have PHP8.2 support. Without that the screen was full of warnings and critical errors. I included the package with composer, so I also had to rename some paths in the code.
I also included a strftime polyfill which seems to work great, see https://github.com/s9y/Serendipity/issues/784

Can we merge the improvements already made in already?
At least it is a good starting point to fix warnings because it is no longer completely broken.

### CODE REVIEW

File: bundled-libs/psr/simple-cache/composer.json [hunk 1]
```diff
@@ -6,11 +6,11 @@
     "authors": [
         {
             "name": "PHP-FIG",
-            "homepage": "http://www.php-fig.org/"
+            "homepage": "https://www.php-fig.org/"
         }
     ],
     "require": {
-        "php": ">=5.3.0"
```
Comment (onli) [inline bundled-libs/psr/simple-cache/composer.json:13]: This change is strange. I see that it was not intended, but how can we solve it? voku/simple-cache has psr/simple-cache as dependency, but sets it to 1.0 || 2.0, with only 2.0 requiring PHP 8.0. 

We should either bundle 1.0 (if compatible with PHP 8), somehow bundle both and use the one that fits, or swap out voku/simple-cache?

**Comment by stevleibelt** [comment]
One general question, why merge to master and not to a feature branch like "update-to-php-8-2"?

To much complexity overhead?

Cheers and thanks for your work.

**Comment by surrim** [comment]
Because I've changed many files and don't want to have big merge conflicts. Nothing is rewritten, so now it's working. If somebody else is making changes, maybe the code is failing later.

File: bundled-libs/psr/simple-cache/composer.json (hunk 1 above)
Comment (surrim) [reply bundled-libs/psr/simple-cache/composer.json:13]: I have no idea, I don't know any of them. I have only updated Smarty and included the strftime module - this was the result.
Fell free to modify the branch ;-)

**Comment by onli** [comment]
@surrim, should we merge this and work on the remaining tasks - like the cache library - in master?

**Pull request merged**